packages/backend-core/src/users/db.ts

```diff
@@ -446,9 +453,32 @@ export class UserDB {
       creator => !!creator
     ).length
 
+    const ssoUsersToDelete: AnyDocument[] = []
     for (let user of usersToDelete) {
+      const platformUser = (await getFirstPlatformUser(
+        user._id!
+      )) as PlatformUserById
```

Comment:
If I am right, this cast should not be needed anymore

Reply:
It is unfortunately because of the line below which assumes the presence of the ssoId. The platform user types are a little strange because of the multiple Couch docs:

```ts
/**
 * doc id is user email
 */
export interface PlatformUserByEmail extends Document {
  tenantId: string
  userId: string
}

/**
 * doc id is userId
 */
export interface PlatformUserById extends Document {
  tenantId: string
  email?: string
  ssoId?: string
}

/**
 * doc id is a unique SSO provider ID for the user
 */
export interface PlatformUserBySsoId extends Document {
  tenantId: string
  userId: string
  email: string
  ssoId?: string
}

export type PlatformUser =
  | PlatformUserByEmail
  | PlatformUserById
  | PlatformUserBySsoId
```